python/modules/GenPartFilter.py

```diff
@@ -108,15 +108,15 @@ def analyze(self, event):
         vMapDict = np.vectorize(mapDict.__getitem__)
         GenPartCut_genPartIdxMother_filtered = vMapDict(GenPartCut_genPartIdxMother[filterArray])
 
-        self.out.fillBranch("GenPartCut_pt",               GenPartCut_pt[filterArray])
-        self.out.fillBranch("GenPartCut_eta",              GenPartCut_eta[filterArray])
-        self.out.fillBranch("GenPartCut_phi",              GenPartCut_phi[filterArray])
-        self.out.fillBranch("GenPartCut_mass",             GenPartCut_mass[filterArray])
+        self.out.fillBranch("GenPart_pt",               GenPartCut_pt[filterArray])
+        self.out.fillBranch("GenPart_eta",              GenPartCut_eta[filterArray])
+        self.out.fillBranch("GenPart_phi",              GenPartCut_phi[filterArray])
+        self.out.fillBranch("GenPart_mass",             GenPartCut_mass[filterArray])
 
-        self.out.fillBranch("GenPartCut_genPartIdxMother", GenPartCut_genPartIdxMother_filtered)
-        self.out.fillBranch("GenPartCut_pdgId",            GenPartCut_pdgId[filterArray])
-        self.out.fillBranch("GenPartCut_status",           GenPartCut_status[filterArray])
-        self.out.fillBranch("GenPartCut_statusFlags",      GenPartCut_statusFlags[filterArray])
+        self.out.fillBranch("GenPart_genPartIdxMother", GenPartCut_genPartIdxMother_filtered)
```

Comment:
Will the mother index brake ? Also, you need to update nGenPart I think.

Reply:
https://github.com/susy2015/NanoSUSY-tools/pull/14/files#diff-d56d9b8c18373b512e908787a3fac852R102

and no, "nGenPart" is handled automatically by the array branches when they are filled.

Reply:
OK， I assume you run the test and things work as it is. Merging this PR to dev2